packages/ruleset/src/utils/validate-subschemas.js

```diff
@@ -0,0 +1,36 @@
+const validateSubschemas = (schema, path, validate) => {
+  const errors = [];
+  // invoke validation function
+  errors.push(...validate(schema, path));
+
+  // recursively process subschemas
+  if (schema.properties) {
+    for (const property of Object.entries(schema.properties)) {
+      errors.push(
+        ...validateSubschemas(
+          property[1],
+          [...path, 'properties', property[0]],
```

Comment:
Since we're building the path ourself here (from what I assume is a "resolved" document, is it safe to assume Spectral will back-solve for the real path from this resolved path?

Reply:
Yes, we return the actual path from the resolved document to Spectral and they will back-solve for the real path when returning results (assuming that the results are coming from a file and not an inline JSON object. I don't know why Spectral requires the filepath to back-solve but they do).